pkg/controller/controller.go

```diff
@@ -1124,6 +1129,23 @@ func (p *csiProvisioner) getSnapshotSource(ctx context.Context, claim *v1.Persis
 		}
 	}
 
+	// New code
+	if p.preventVolumeModeConversion {
+		if snapContentObj.Spec.SourceVolumeMode != nil && snapContentObj.Spec.SourceVolumeMode != claim.Spec.VolumeMode {
+			allowVolumeModeChange, ok := snapContentObj.Annotations[annAllowVolumeModeChange]
+			if !ok {
+				klog.Infof("Raunak1")
+			}
+			allowVolumeModeChangeBool, err := strconv.ParseBool(allowVolumeModeChange)
+			if err != nil {
+				klog.Infof("Raunak2")
+			}
+			if !allowVolumeModeChangeBool {
+				klog.Infof("Raunak3")
```

Comment:
Those are debug messages with your name in it?

Reply:
@xing-yang my apologies, this PR is not ready for review. Let me make it explicit in the PR description! I'm still testing and opened a PR to get the images built